docs/model-dev-guide/apis-howto/deepspeed/autotuning.rst

```diff
@@ -0,0 +1,302 @@
+.. _deepspeed-autotuning:
+
+################################
+ DeepSpeed Autotune: User Guide
+################################
+
+.. meta::
+   :description: This user guide demonstrates how to optimize DeepSpeed parameter in order to take full advantage of the user's hardware and model.
+
+Getting the most out of DeepSpeed (DS) requires aligning the many DS parameters with the specific
+properties of your hardware and model. Determined AI's DeepSpeed Autotune (``dsat``) helps to
+optimize these settings through an easy-to-use API with very few changes required in user-code, as
+we describe in the remainder of this user guide. ``dsat`` can be used with
+:class:`~determined.pytorch.deepspeed.DeepSpeedTrial`, :ref:`Core API <core-getting-started>`, and
+`HuggingFace Trainer <https://huggingface.co/docs/transformers/main_classes/trainer>`__.
+
+**************
+ How it Works
+**************
+
+You do not need to create a special configuration file to use ``dsat``. Assuming you have DeepSpeed
+code which already functions, autotuning is as easy as inserting one or two helper functions into
+your code and modifying the launch command.
+
+For instance, let us imagine that your present directory contains DeepSpeed code and a corresponding
+``single`` trial experiment configuration file ``deepspeed.yaml``. Then, after inserting a line or
```

Comment:
For instance, let's say your directory contains DeepSpeed code and a corresponding...